### PR TITLE
Add support to disable JITServer and AOT post restore

### PR DESCRIPTION
**Pull request by dsouzai**
Since JITServer is enabled implicitly in Non Portable Mode, specifying `-XX:-UseJITServer` should disable it post-restore.

Additionally, if `-Xshareclasses:disableOnRestore` is specified, disable AOT (see https://github.com/eclipse-openj9/openj9/issues/16868).

### CODE REVIEW

**Comment by dsouzai** [comment]
Converting to draft so I can also add the change to support disabling the SCC post restore in this PR (to minimize the amount of double delivering needed before Friday).

File: runtime/compiler/control/OptionsPostRestore.cpp [hunk 1]
```diff
@@ -261,6 +263,13 @@ J9::OptionsPostRestore::iterateOverExternalOptions()
             }
             break;
 
+         case J9::ExternalOptions::XShareclassesDisableOnRestore:
+            {
+            if (FIND_ARG_IN_RESTORE_ARGS(OPTIONAL_LIST_MATCH, optString, 0) >= 0)
```
Comment (dsouzai) [inline runtime/compiler/control/OptionsPostRestore.cpp:268]: At the moment, running with `-Xshareclasses:disableOnRestore` will fail on restore (unless I change this to `FIND_AND_CONSUME_RESTORE_ARGS`). I'm expecting the VM to use the `FIND_AND_CONSUME` variant when they process this option.

fyi @hangshao0

**Comment by dsouzai** [comment]
@mpirvu ready for review now.

**Review by mpirvu — APPROVED** [review]
LGTM
I wonder if we should reverse the logic for JITServer: run in client mode before checkpoint (as we do now), but by default disable client-mode post restore, unless the user specifies -XX:+UseJITServer in which case we let the JVM in client mode.
This would be more in line with current behavior where the user has to opt-in to use JITServer.

**Comment by mpirvu** [comment]
jenkins test sanity all jdk17

**Comment by dsouzai** [comment]
> I wonder if we should reverse the logic for JITServer: run in client mode before checkpoint (as we do now), but by default disable client-mode post restore, unless the user specifies -XX:+UseJITServer in which case we let the JVM in client mode.
This would be more in line with current behavior where the user has to opt-in to use JITServer.

It's something we can look into for post 0.38; I have a task in https://github.com/eclipse-openj9/openj9/issues/16853 under the `JITServer` section to support `-XX:+UseJITServer` post restore in Portable mode which maybe could make use of this same idea.

That said, I believe having JITServer enabled implicitly in Non-Portable mode was a deliberate decision, though it was made before we had the ability to specify options post-restore, so it could be something to revisit. @vijaysun-omr @ymanton what are your thoughts on Marius' suggestion (since you guys made the decision to implicitly enable JITServer).

**Pull request merged**